### ISSUE TITLE
Error passing any_int/any_float as sum type function argument

### ISSUE THREAD

**Issue by endeveit** [vlang/v#5952]
**V version:** `V 0.1.28 53e7cb1.8f23acc`
**OS:** `macOS 10.15.5`

**What did you do?**
I'm using the sum type to build a similar logic for different types, here's the example of the code:
```v
module main

type Number = int | f64

fn is_gt(val string, dst Number) bool {
	match dst {
		int {
			return val.int() > *dst
		}
		f64 {
			return val.f64() > *dst
		}
	}
}

fn main() {
	rs := is_gt('3', 2)
	println(rs)
}
```

**What did you expect to see?**
The program works as expected:
```bash
$ v run main.v
true
```
 
**What did you see instead?**
A compilation error:
```bash
$ v run main.v
main.v:17:8: error: cannot use type `any_int` as type `Number` in argument 2 to `is_gt`
   15 |
   16 | fn main() {
   17 |     rs := is_gt('3', 2)
      |           ~~~~~~~~~~~~~
   18 |     println(rs)
   19 | }
```
On Discord #help channel `Delta456` suggested that this is due to the following:
```
>>> typeof(2)
any_int
```

**Comment by ntrel** [comment]
This works:
```
	i := 2
	rs := is_gt('3', i)
	println(rs)
	f := 3.3
	r2 := is_gt('3', f)
	println(r2)
```
`is_gt('3', 3.3)` does not work - error: cannot use type `any_float` as type `Number` in argument 2 to `is_gt`.

**Comment by ntrel** [comment]
Also V should allow `> dst`, not `> *dst`, but that's a separate issue I suppose.

**Comment by UweKrueger** [comment]
A literal like `2` is a valid `int`, a valid `f64`, a valid `u64`, ...
If there has to be made a decision what `2` is as a sum type, this decision cannot be made - so the compiler is right to report an error.
It's not a bug - it's a feature.

**Comment by UweKrueger** [comment]
This comes up too often, so I changed my mind... I'll implement suitable default conversions. See #6575

**Comment by UweKrueger** [comment]
This should now be fixed in bf904c2 by #7864.

**Comment by endeveit** [comment]
Thanks, @UweKrueger!
